utils/filesystem/files_test.go

```diff
@@ -1306,6 +1306,37 @@ func TestFilepathStem(t *testing.T) {
 	})
 }
 
+func TestUnzipFileCountLimit(t *testing.T) {
+	fs := NewFs(StandardFS)
+
+	testInDir := "testdata"
```

Comment:
for ease of readiness, you should create subtest so that the reporting gets clear
i.e.
```
t.Run("zip under the limits", func(){
....
})
```
then
```
t.Run("zip above the limits", func(){
....
})
```